src/suites/cts/validation/createBindGroupLayout.spec.ts

```diff
@@ -148,3 +148,91 @@ g.test('dynamic set to true is allowed only for buffers', async t => {
   { type: 'sampled-texture', _success: false },
   { type: 'storage-texture', _success: false },
 ]);
+
+g.test('number of resources per stage exceeds maximum value for resource type', async t => {
+  const { type, maximumCount } = t.params;
+
+  const maxResourceBindings: GPUBindGroupLayoutBinding[] = [];
+  for (let i = 0; i < maximumCount; i++) {
+    maxResourceBindings.push({
+      binding: i,
+      visibility: GPUShaderStage.FRAGMENT,
+      type
+    });
+  }
+
+  const goodDescriptor: GPUBindGroupLayoutDescriptor = { bindings: maxResourceBindings };
+
+  // Control
+  t.device.createBindGroupLayout(goodDescriptor);
+
+  const badDescriptor = clone(goodDescriptor);
+  badDescriptor.bindings.push({
+    binding: maximumCount,
+    visibility: GPUShaderStage.FRAGMENT,
```

Comment:
... here, use VERTEX, FRAGMENT, or COMPUTE based on a parameter.

You can easily add this to the existing params list using `pcombine` (cartesian product) with `poptions('visibility', ['VERTEX', 'FRAGMENT', 'COMPUTE']`.